cmd/snap-confine/apparmor-support.c

```diff
@@ -100,6 +104,21 @@ void sc_init_apparmor_support(struct sc_apparmor *apparmor)
 	} else {
 		apparmor->mode = SC_AA_INVALID;
 	}
+
+	// Check that apparmor is actually usable. In some configurations
+	// of lxd apparmor looks available but it is not usable. This is
+        // the case in when a container runs unprivileged and also
+        // unconfined.
```

Comment:
There is a whitespace issue here. I might make sense to expand on the comment so we remember for next time. I suggest:
```
// Check that apparmor is actually usable. In some configurations
// of lxd, apparmor looks available when in reality it isn't. Eg, this
// can happen when a container runs unprivileged (eg, root in the
// container is non-root outside) and also unconfined (where lxd
// doesn't set up an apparmor policy namespace). We can therefore
// simply check if /sys/kernel/security/apparmor/profiles is readable
// (like aa-status does), and it isn't, we know we can't manipulate
// policy.
```